### DISCUSSION TITLE
Allow for folding between matching expressions

### DISCUSSION THREAD

**Discussion by jason-pollock** [krasa/GrepConsole#239] [Ideas]
I like the "Use a previous highlight if a line is not matched" feature, and would love to have something similar for Folds. For example, I'd like to fold all lines between `>>>` and `<<<`.

```
>>>
These
lines
should
all
be
folded
<<<
```

**Comment by krasa** [comment]
Thank you for the donation. 

Sounds good, how does it look?
![image](https://user-images.githubusercontent.com/1160875/184023738-e0907509-2fa7-42ff-8df8-e55b98d8b0ff.png)

There is a limitation of the API, it is not possible to separate folded lines that are next to each other into more foldings, but it is possible to not include the closing line:
![image](https://user-images.githubusercontent.com/1160875/184024700-bcca4162-c3ae-43e9-8d96-69ed07365407.png)

**Reply by jason-pollock** [reply]
This is a perfectly awesome solution! But my Grep Console window doesn't have the "Folding" section shown in your screenshots so I'm unable to replicate your settings. I'm running PHPStorm 2022.2 with Grep Console v12.14.211.6693.0

I've checked PHPStorm for Mac and Linux, and I can't figure out how to enable the "Folding" section on either one.

![CleanShot 2022-08-11 at 08 13 31](https://user-images.githubusercontent.com/85717231/184141583-5119097d-7559-4e7e-ab83-12bcf7ec5e1b.png)

**Comment by krasa** [comment]
because I just implemented it :-)

**Comment by krasa** [comment]
Thanks again, dunno why you got that error. 
You can try it: [GrepConsole-12.15.211.6693.0.zip](https://github.com/krasa/GrepConsole/files/9309349/GrepConsole-12.15.211.6693.0.zip) (install the whole zip)

**Comment by krasa** [comment]
it was released

**Reply by jason-pollock** [reply]
Sorry for the delay - this works perfectly! Thank you for the feature.
